src/HealthChecks.AzureDigitalTwin/AzureDigitalTwinModelsHealthCheck.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Azure.Core;
+using Azure.DigitalTwins.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HealthChecks.AzureDigitalTwin
+{
+    public class AzureDigitalTwinModelsHealthCheck
+           : IHealthCheck
+    {
+        private static readonly ConcurrentDictionary<string, DigitalTwinsClient> _digitalTwinClientConnections
+            = new ConcurrentDictionary<string, DigitalTwinsClient>();
+
+        private readonly string _clientId;
+        private readonly string _clientSecret;
+        private readonly string _tenantId;
+        private readonly TokenCredential _credentials;
+        private readonly string _hostName;
+        private readonly string[] _models;
+
+        public AzureDigitalTwinModelsHealthCheck(string clientId, string clientSecret, string tenantId, string hostName, string[] models)
+        {
+            _clientId = (!string.IsNullOrEmpty(clientId)) ? clientId : throw new ArgumentNullException(nameof(clientId));
+            _clientSecret = (!string.IsNullOrEmpty(clientSecret)) ? clientSecret : throw new ArgumentNullException(nameof(clientSecret));
+            _tenantId = (!string.IsNullOrEmpty(tenantId)) ? tenantId : throw new ArgumentNullException(nameof(tenantId));
+            _credentials = new ClientSecretCredential(_tenantId, _clientId, _clientSecret);
+
+            _hostName = (!string.IsNullOrEmpty(hostName)) ? hostName : throw new ArgumentNullException(nameof(hostName));
+            _models = models.Any() ? models : throw new ArgumentNullException(nameof(models));
+        }
+
+        public AzureDigitalTwinModelsHealthCheck(TokenCredential credentials, string hostName, string[] models)
+        {
+            _credentials = (credentials is not null) ? credentials : throw new ArgumentNullException(nameof(credentials));
+
+            _hostName = (!string.IsNullOrEmpty(hostName)) ? hostName : throw new ArgumentNullException(nameof(hostName));
+            _models = models.Any() ? models : throw new ArgumentNullException(nameof(models));
```

Comment:
`ArgumentNullException` looks strange if one passes empty array.

Reply:
For me it has no sense if a user try to compare/check for an empty DT but, it's possible that someone wants to be sure that has an empty initial models, so .... validation removed!